src/System.Net.Security/src/System/Net/_SecureChannel.cs

```diff
@@ -0,0 +1,1487 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Security.Authentication.ExtendedProtection;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Globalization;
+using System.ComponentModel;
+using System.Security.Principal;
+using System.Security;
+using System.Collections;
+using System.Diagnostics;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.Security
+{
+    //
+    // SecureChannel - a wrapper on SSPI based functionality. 
+    // Provides an additional abstraction layer over SSPI for SslStream.
+    //
+    internal class SecureChannel
+    {
+        //also used as a lock object
+        internal const string SecurityPackage = "Microsoft Unified Security Protocol Provider";
+        private static readonly object s_SyncObject = new object();
+
+        private const Interop.Secur32.ContextFlags RequiredFlags =
+            Interop.Secur32.ContextFlags.ReplayDetect |
+            Interop.Secur32.ContextFlags.SequenceDetect |
+            Interop.Secur32.ContextFlags.Confidentiality |
+            Interop.Secur32.ContextFlags.AllocateMemory;
+
+        private const Interop.Secur32.ContextFlags ServerRequiredFlags =
+            RequiredFlags | Interop.Secur32.ContextFlags.AcceptStream;
+
+        private const int ChainRevocationCheckExcludeRoot = 0x40000000;
+
+        // When reading a frame from the wire first read this many bytes for the header.
+        internal const int ReadHeaderSize = 5;
+
+        private static volatile X509Store s_MyCertStoreEx;
+        private static volatile X509Store s_MyMachineCertStoreEx;
```

Comment:
Naming: A couple s_PascalCased statics exist in this type (as opposed to s_camelCase)